### DISCUSSION TITLE
Interesting - encoder and potentiometer modules that may be useful with MiniDexed

### DISCUSSION THREAD

**Discussion by pik33** [probonopd/MiniDexed#477] [Ideas]
I found these things. They may be useful for controlling all kind of DIY synths instead of connecting discrete encoders. 

8 encoders or 8 potentiometers, with a controller, communicating via i2c. 

https://shop.m5stack.com/products/8-encoder-unit-stm32f030?variant=43585053851905
https://shop.m5stack.com/collections/m5stack-new-arrival/products/8-angle-unit-with-potentiometer

**Comment by mongio** [comment]
Wow, if they could be used with a Disting EX eurorack module, for example...look very useful and cheap.

**Comment by diyelectromusic** [comment]
Funnily enough, I've just written about using the pot one as a MIDI controller by having it plugged into a XIAO.,

https://diyelectromusic.wordpress.com/2023/04/12/xiao-samd21-arduino-and-midi-part-7/

I plan to do the encoder one next, but am not quite sure how to map it onto MIDI CCs.  What do people tend to do with encoders that might be useful?

Kevin

**Reply by diyelectromusic** [reply]
And I've now updated the post to show either the 8Angle or 8Encoder as an 8-way MIDI CC controller when used with the XIAO and a XIAO expander board.

Kevin

![IMG_7101](https://user-images.githubusercontent.com/68612569/231801671-80e37af5-2503-4c7a-ad94-c5ccbc3cf711.JPG)

**Comment by ghost** [comment]
Here is rough-hewn home made prototype, based on a Raspberry Pico. I haven't tried it with MiniDexed yet. The joystick is mapped to the standard midi ccs for pitchbend and mod wheel; the rotary encoders and faders are mapped to unallocated ccs. I use it with soft synths that have a midi learn function.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by diyelectromusic** [reply]
I'm never quite sure with encoders and CCs.  pots make perfect sense, a definite start/end point mapped onto 0..127, but a continuous encoder, I'm not really sure about.  I've mapped encoders to the same range but with wrap around, which made sense to me :)

Kevin

**Comment by ghost** [comment]
![IMG-20230409-WA0001](https://user-images.githubusercontent.com/103966073/231831986-64ddfd34-eb72-4305-89a8-22c87ede1b0f.jpg)
Here it is. Needs a bit of work before production...

**Comment by ghost** [comment]
I struggled with the encoders for a while. It's easy to limit the sent value to the range 1 to 127, but the encoder tends to 'wind on up' (or down) if you keep turning it.

It's a bit like driving a car, and the steering wheel keeps turning even after you reach full lock!  However it is possible to code them to behave like a pot.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by diyelectromusic** [reply]
Oh indeed.  I'm just not sure, given the increased complexity of using an encoder, why one wouldn't just use pots for MIDI :)

But I expect they have a use somewhere!

Kevin

**Reply by pik33** [reply]
I have a midi keyboard with potentiometers. They changes the output values while playing (mechanical vibrations) and this means they can be used only for volume control, when changing the parameter doesn't matter that much. They can not be used for (for example) program change (a funny effect occurs while playing). A side effect is a lot unnecessary midi reports send by the keyboard from these potentiometer controlled MIDI control channels

Encoders have no such effect.
